### ISSUE TITLE
Vertical Axis Values are clipped in scrollable chart 

### ISSUE THREAD

**Issue by roly151** [infinum/flutter-charts#63]
I have a scrollable chart and want to show vertical axis values from index. However these are clipped as the size allowed for the value is too small for the value from index - unless I make the width of the chart really wide. The values in the chart image below should be 12am, 5am, 10am etc.

![Screen Shot 2022-09-23 at 4 45 48 pm](https://user-images.githubusercontent.com/27068792/191905573-d5bb7312-f30a-4e49-a4be-e259dbf58786.png)

If I make the text smaller, it shows more of the values. 

![Screen Shot 2022-09-23 at 5 25 55 pm](https://user-images.githubusercontent.com/27068792/191910816-d2580942-574b-4501-9bf6-fdc999d7c1dc.png)

Or if I make the chart width really wide, then I can get more of the values showing. 

![Screen Shot 2022-09-23 at 5 35 29 pm](https://user-images.githubusercontent.com/27068792/191912237-b5728856-1832-4b41-a0f5-76d2d1d1c5ff.png)

But how can I get the values to show without making the chart really wide? If the verticalAxisStep is set, is it possible to allow the values to show past the allowed space without clipping them? (Hope that makes sense)

The code for this chart is:

```
Expanded(
          child: SingleChildScrollView(
            scrollDirection: Axis.horizontal,
            child: SizedBox(
              width: 160 * SizeConfig.safeBlockHorizontal,
              height: widget.chartHeight,
              child: 
                  Chart(
                state: _chartState,
              ),
            ),
          ),
        ),
```

``` 
_chartState = ChartState.line(
      ChartData.fromList(
        temperatures.map((e) => BubbleValue<void>(e.toDouble())).toList(),
        axisMax: maxY,
        axisMin: minY,
      ),
      behaviour: const ChartBehaviour(
        isScrollable: true,
      ),
      backgroundDecorations: [
        GridDecoration(
          showVerticalValues: true,
          showHorizontalValues: false,
          verticalAxisStep: 5,
          gridColor: CustomColors.darkBlue.withOpacity(0),
          textStyle: const TextStyle(
            color: CustomColors.darkBlue,
            fontSize: 14,
          ),
          verticalAxisValueFromIndex: (index) {
            var now = DateTime.now();
            var midnight = DateTime(now.year, now.month, now.day);
            var time = DateFormat('h').format(midnight.add(
                  Duration(minutes: index.toInt() * 60),
                )) +
                DateFormat('a')
                    .format(midnight.add(
                      Duration(minutes: index.toInt() * 60),
                    ))
                    .toLowerCase();
            return time;
          },
        ),
      ],
      foregroundDecorations: [
        SparkLineDecoration(
          lineWidth: 2.0,
          gradient: lineColor(minY, maxY),
          smoothPoints: true,
        ),
      ],
    );
```

**Comment by lukaknezic** [comment]
Hey @roly151, thanks for report 😄 

There was issue that text painter would cut off text on single item width. So even though you set step size to 5, text painter didn't take that into account.

Now that is fixed and text should also be centred as well. (You can change alignment)
You can find fix in this PR: #64

**Comment by itsJoKr** [comment]
Since there's no additional comments, and we have a fix I will close this.
